mm/2s2h/Rando/ActorBehavior/EnemyDrops.cpp

```diff
@@ -0,0 +1,148 @@
+    { ACTOR_EN_SB, RC_ENEMY_DROP_SHELLBLADE },
+    { ACTOR_EN_PR2, RC_ENEMY_DROP_SKULLFISH },
+    { ACTOR_EN_ST, RC_ENEMY_DROP_SKULLTULA },
+    { ACTOR_EN_SW, RC_ENEMY_DROP_SKULLWALLTULA },
+    { ACTOR_EN_KAME, RC_ENEMY_DROP_SNAPPER },
+    { ACTOR_EN_SKB, RC_ENEMY_DROP_STALCHILD },
+    { ACTOR_EN_TITE, RC_ENEMY_DROP_TEKTITE },
+    { ACTOR_EN_WF, RC_ENEMY_DROP_WOLFOS },
+    { ACTOR_EN_WALLMAS, RC_ENEMY_DROP_WALLMASTER },
+};
+
+RandoCheckId GetRandoCheckByActorId(int16_t actorId) {
+    for (auto& map : actorIdToRandoCheckIdMap) {
+        if (map.first == actorId) {
+            return map.second;
+        }
+    }
+
+    return RC_UNKNOWN;
+}
+
+Actor* FindActor(Vec3f position) {
+    ActorListEntry actorList = gPlayState->actorCtx.actorLists[ACTORCAT_ENEMY];
+
+    Actor* currentActor = actorList.first;
+    for (size_t i = 0; i < actorList.length; i++) {
+        if (currentActor->world.pos.x != position.x && currentActor->world.pos.z != position.z) {
+            currentActor = currentActor->next;
+        } else {
+            return currentActor;
+        }
+    }
+    return nullptr;
+}
+
+void SpawnEnemyDrop(Vec3f position, Actor* actor, RandoCheckId randoCheckId) {
+    CustomItem::Spawn(
+        position.x, position.y, position.z, 0,
+        CustomItem::KILL_ON_TOUCH | CustomItem::TOSS_ON_SPAWN | CustomItem::ABLE_TO_ZORA_RANG, randoCheckId,
+        [](Actor* actor, PlayState* play) {
+            RandoSaveCheck& randoSaveCheck = RANDO_SAVE_CHECKS[CUSTOM_ITEM_PARAM];
+            randoSaveCheck.eligible = true;
+        },
+        [](Actor* actor, PlayState* play) {
+            auto& randoSaveCheck = RANDO_SAVE_CHECKS[CUSTOM_ITEM_PARAM];
+            RandoItemId randoItemId = Rando::ConvertItem(randoSaveCheck.randoItemId);
+            Matrix_Scale(30.0f, 30.0f, 30.0f, MTXMODE_APPLY);
+            Rando::DrawItem(Rando::ConvertItem(randoSaveCheck.randoItemId, (RandoCheckId)CUSTOM_ITEM_PARAM), actor);
+        });
+}
+
+void Rando::ActorBehavior::InitEnemyDropBehavior() {
+    COND_VB_SHOULD(VB_ENEMY_DROP_COLLECTIBLE, IS_RANDO, {
+        Vec3f position = va_arg(args, Vec3f);
+
+        if (RANDO_SAVE_OPTIONS[RO_SHUFFLE_ENEMY_DROPS] == RO_GENERIC_OFF) {
+            return;
+        }
+
+        Actor* foundActor = FindActor(position);
+        if (foundActor == nullptr || foundActor->category != ACTORCAT_ENEMY) {
+            return;
+        }
+
+        if (foundActor->id == ACTOR_EN_SW) {
+            return;
+        }
+
+        RandoCheckId randoCheckId = GetRandoCheckByActorId(foundActor->id);
+        if (RANDO_SAVE_CHECKS[randoCheckId].obtained) {
+            return;
+        }
+
+        SpawnEnemyDrop(position, foundActor, randoCheckId);
+        *should = false;
+    });
+
+    COND_ID_HOOK(OnActorKill, ACTOR_EN_SW, IS_RANDO, [](Actor* actor) {
+        SPDLOG_INFO("Actor ID: {} | Actor Params: {}", std::to_string(actor->id), std::to_string(actor->params));
+        if (actor->params == -4) {
```

Comment:
I don't think this is apt. It seems like the intent is to check that this is a Skullwalltula, not a gold Skulltula. In which case, it should be:

```suggestion
        if (!ENSW_GET_3(actor)) { // Is a regular Skullwalltula
```